appsmithorg/appsmith
# fix: updated logic to retain the files on page change

## Description

**PR Changes context:**
- To retain the files when the user returns to the canvas from query page or any page, we initialize the uppy's instance with the files that are present in the `selectedFiles` meta property. 
- During this process, we also add a flag inside the file's meta called: `isInitializing`. This helps us to prevent adding same files with different `file.id`. Note: during initialization `addFile` function will generate a different `file.id` for the files present in the `selectedFiles` meta property.
- We initialze the files returned by the `getFiles` function whenever a file is removed by a user.

Fixes #15002
Fixes #11660

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Updated the cypress test: `FilePickerV2_spec.js` to check the following scenario:
* Upload a file to the file picker.
* Check if the uploaded file's name reflects in the text widget.
* Goes to the query page, and checks if the binding `{{FilePicker1.files}}` evaluates to the file picker's file content. Here we check if the file name appears in the Evaluation pop-up.
* Goes back to the canvas, checks if the uploaded file is retained by checking if the file picker button contains: `1 files selected`.

## Checklist:
